src/View/SSViewer.php

```diff
@@ -57,210 +36,98 @@ class SSViewer implements Flushable
     /**
      * A list (highest priority first) of themes to use
      * Only used when {@link $theme_enabled} is set to TRUE.
-     *
-     * @config
-     * @var string
      */
-    private static $themes = [];
-
-    /**
-     * Overridden value of $themes config
-     *
-     * @var array
-     */
-    protected static $current_themes = null;
+    private static array $themes = [];
 
     /**
      * Use the theme. Set to FALSE in order to disable themes,
      * which can be useful for scenarios where theme overrides are temporarily undesired,
      * such as an administrative interface separate from the website theme.
      * It retains the theme settings to be re-enabled, for example when a website content
      * needs to be rendered from within this administrative interface.
-     *
-     * @config
-     * @var bool
      */
-    private static $theme_enabled = true;
+    private static bool $theme_enabled = true;
 
     /**
-     * Default prepended cache key for partial caching
-     *
-     * @config
-     * @var string
+     * If true, rendered templates will include comments indicating which template file was used.
+     * May not be supported for some rendering engines.
      */
-    private static $global_key = '$CurrentReadingMode, $CurrentUser.ID';
+    private static bool $source_file_comments = false;
 
     /**
-     * @config
-     * @var bool
+     * Set if hash links should be rewritten
      */
-    private static $source_file_comments = false;
+    private static bool $rewrite_hash_links = true;
 
     /**
-     * Set if hash links should be rewritten
-     *
-     * @config
-     * @var bool
+     * Overridden value of $themes config
      */
-    private static $rewrite_hash_links = true;
+    protected static array $current_themes = [];
 
     /**
      * Overridden value of rewrite_hash_links config
      *
-     * @var bool
+     * Can be set to "php" to rewrite hash links with PHP executable code.
      */
-    protected static $current_rewrite_hash_links = null;
+    protected static null|bool|string $current_rewrite_hash_links = null;
```

Comment:
presume the null is because we want a tri-state.  Though string? why string?

Reply:
Because that was already the case. I'm just adding the type that represents what's already happening, not refactoring the way hash link rewriting gets configured. See the `process()` method which has a `if ($rewrite === 'php') {` condition.